<!DOCTYPE html>





    

    

    

    

<html lang="en-us"><head><script src="/bdi-viz-manual/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=bdi-viz-manual/livereload" data-no-instant defer></script>
    <meta charset="utf-8" />
    <title>Dataset to GDC Schema Alignment | BDIViz User Manual</title>
    <meta name="robots" content="noindex">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="Comprehensive guide for mapping biomedical datasets to the GDC schema standard">
    <meta name="keywords" content="Documentation, Hugo, Hugo Theme, Bootstrap" />
    <meta name="author" content="Colin Wilson - Lotus Labs" />
    <meta name="email" content="support@aigis.uk" />
    <meta name="website" content="https://lotusdocs.dev" />
    <meta name="Version" content="v0.1.0" />
    
    <link rel="icon" href="http://localhost:1313/bdi-viz-manual/favicon.ico" sizes="any">
<link rel="icon" type="image/svg+xml" href="http://localhost:1313/bdi-viz-manual/favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="http://localhost:1313/bdi-viz-manual/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="http://localhost:1313/bdi-viz-manual/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="http://localhost:1313/bdi-viz-manual/favicon-16x16.png">
<link rel="manifest" crossorigin="use-credentials" href="http://localhost:1313/bdi-viz-manual/site.webmanifest">
<meta property="og:title" content="Dataset to GDC Schema Alignment" />
<meta property="og:description" content="Comprehensive guide for mapping biomedical datasets to the GDC schema standard" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/bdi-viz-manual/docs/get-started/dataset-to-gdc/" /><meta property="og:image" content="http://localhost:1313/bdi-viz-manual/opengraph/card-base-2_hu_7c30a31a7dccdb84.png"/><meta property="article:section" content="docs" />
<meta property="article:published_time" content="2025-04-19T13:38:17-04:00" />
<meta property="article:modified_time" content="2025-04-19T13:38:17-04:00" /><meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="http://localhost:1313/bdi-viz-manual/opengraph/card-base-2_hu_7c30a31a7dccdb84.png"/>
<meta name="twitter:title" content="Dataset to GDC Schema Alignment"/>
<meta name="twitter:description" content="Comprehensive guide for mapping biomedical datasets to the GDC schema standard"/>

    
        <link rel="alternate" type="application/atom+xml" title="Atom feed for BDIViz User Manual" href="/index.xml" />
    
    
    
            
                <script type="text/javascript" src="http://localhost:1313/bdi-viz-manual/docs/js/flexsearch.bundle.js"></script>
            
        
    
    
    
    
        
        
        
        
    
        
        
        
        
    
    
    <link rel="preconnect" href="https://fonts.gstatic.com/" />
    <link rel="preconnect" href="https://fonts.gstatic.com/" crossorigin />
    <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,600,700|Helvetica:300,400,600,700&display=block" rel="stylesheet">

    <link rel="stylesheet" href="/bdi-viz-manual/docs/scss/style.css" crossorigin="anonymous">
    
    
    </head>
<body>
        <div class="content">
            <div class="page-wrapper toggled">
<nav id="sidebar" class="sidebar-wrapper">
    <div class="sidebar-brand">
        <a href='/bdi-viz-manual' aria-label="HomePage" alt="HomePage">
            
                <?xml version="1.0" encoding="UTF-8"?><svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 250 250"><path d="m143,39.5c-18,0-18,18-18,18,0,0,0-18-18-18H22c-2.76,0-5,2.24-5,5v143c0,2.76,2.24,5,5,5h76c7.2,0,8.64,11.52,8.93,16.13.07,1.05.95,1.87,2,1.87h32.14c1.06,0,1.94-.82,2-1.87.29-4.61,1.73-16.13,8.93-16.13h76c2.76,0,5-2.24,5-5V44.5c0-2.76-2.24-5-5-5h-85Zm63,123.5c0,1.38-1.12,2.5-2.5,2.5h-60.5c-18,0-18,18-18,18,0,0,0-18-18-18h-60.5c-1.38,0-2.5-1.12-2.5-2.5v-94c0-1.38,1.12-2.5,2.5-2.5h51.5c7.2,0,8.64,11.52,8.93,16.13.07,1.05.95,1.87,2,1.87h32.14c1.06,0,1.94-.82,2-1.87.29-4.61,1.73-16.13,8.93-16.13h51.5c1.38,0,2.5,1.12,2.5,2.5v94Z" style="fill:#06f;"/></svg>
            
        </a>
    </div>
    <div class="sidebar-content" style="height: calc(100% - 131px);">
        <ul class="sidebar-menu">
            
                
                
                    
                        
                        <li class="">
                            <a class="sidebar-root-link" href="http://localhost:1313/bdi-viz-manual/docs/overview/">
                                
                                Overview
                            </a>
                        </li>
                    
                
                    
                    
                        <li class="sidebar-dropdown no-icon current active">
                            <button class="btn">
                                Get Started
                            </button>
                            <div class="sidebar-submenu d-block">
                                <ul>
                                    
                                        
                                        
                                            <li class="current no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/get-started/dataset-to-gdc/">Dataset to GDC Schema Alignment</a></li>
                                        
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/get-started/verify-value-mappings/">Verify Value Matching</a></li>
                                        
                                    
                                </ul>
                            </div>
                        </li>
                    
                
                    
                    
                        <li class="sidebar-dropdown no-icon ">
                            <button class="btn">
                                Installation
                            </button>
                            <div class="sidebar-submenu ">
                                <ul>
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/installation/install-on-linux/">Install on Linux/Unix</a></li>
                                        
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/installation/install-on-macos/">Install on MacOS</a></li>
                                        
                                    
                                </ul>
                            </div>
                        </li>
                    
                
                    
                    
                        <li class="sidebar-dropdown no-icon ">
                            <button class="btn">
                                User Manual
                            </button>
                            <div class="sidebar-submenu ">
                                <ul>
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/manual/upload-your-data/">Upload Your Data</a></li>
                                        
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/manual/explore-matches/">Explore Matches</a></li>
                                        
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/manual/llm-agent/">LLM Agent</a></li>
                                        
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/manual/user-operation/">Perform User Operations</a></li>
                                        
                                    
                                        
                                        
                                            <li class=" no-icon"><a class="sidebar-nested-link" href="http://localhost:1313/bdi-viz-manual/docs/manual/export-your-mappings/">Export Your Mappings</a></li>
                                        
                                    
                                </ul>
                            </div>
                        </li>
                    
                
            
        </ul>
        
    </div>
    
        <ul class="sidebar-footer list-unstyled mb-0">
            
        </ul>
    
</nav>

                    <main class="page-content bg-transparent">
                        
<div id="top-header" class="top-header d-print-none">
    <div class="header-bar d-flex justify-content-between">
        <div class="d-flex align-items-center">
            <a href='/bdi-viz-manual' class="logo-icon me-3" aria-label="HomePage" alt="HomePage">
                <div class="small">
                    
                            <?xml version="1.0" encoding="UTF-8"?><svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 250 250"><path d="m143,39.5c-18,0-18,18-18,18,0,0,0-18-18-18H22c-2.76,0-5,2.24-5,5v143c0,2.76,2.24,5,5,5h76c7.2,0,8.64,11.52,8.93,16.13.07,1.05.95,1.87,2,1.87h32.14c1.06,0,1.94-.82,2-1.87.29-4.61,1.73-16.13,8.93-16.13h76c2.76,0,5-2.24,5-5V44.5c0-2.76-2.24-5-5-5h-85Zm63,123.5c0,1.38-1.12,2.5-2.5,2.5h-60.5c-18,0-18,18-18,18,0,0,0-18-18-18h-60.5c-1.38,0-2.5-1.12-2.5-2.5v-94c0-1.38,1.12-2.5,2.5-2.5h51.5c7.2,0,8.64,11.52,8.93,16.13.07,1.05.95,1.87,2,1.87h32.14c1.06,0,1.94-.82,2-1.87.29-4.61,1.73-16.13,8.93-16.13h51.5c1.38,0,2.5,1.12,2.5,2.5v94Z" style="fill:#06f;"/></svg>
                    
                </div>
                <div class="big">
                    
                            <?xml version="1.0" encoding="UTF-8"?><svg id="Layer_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 250 250"><path d="m143,39.5c-18,0-18,18-18,18,0,0,0-18-18-18H22c-2.76,0-5,2.24-5,5v143c0,2.76,2.24,5,5,5h76c7.2,0,8.64,11.52,8.93,16.13.07,1.05.95,1.87,2,1.87h32.14c1.06,0,1.94-.82,2-1.87.29-4.61,1.73-16.13,8.93-16.13h76c2.76,0,5-2.24,5-5V44.5c0-2.76-2.24-5-5-5h-85Zm63,123.5c0,1.38-1.12,2.5-2.5,2.5h-60.5c-18,0-18,18-18,18,0,0,0-18-18-18h-60.5c-1.38,0-2.5-1.12-2.5-2.5v-94c0-1.38,1.12-2.5,2.5-2.5h51.5c7.2,0,8.64,11.52,8.93,16.13.07,1.05.95,1.87,2,1.87h32.14c1.06,0,1.94-.82,2-1.87.29-4.61,1.73-16.13,8.93-16.13h51.5c1.38,0,2.5,1.12,2.5,2.5v94Z" style="fill:#06f;"/></svg>
                    
                </div>
            </a>
            <button id="close-sidebar" class="btn btn-icon btn-soft">
                <span class="material-icons size-20 menu-icon align-middle">menu</span>
            </button>
            
            
                    
                    <button id="flexsearch-button" class="ms-3 btn btn-soft" data-bs-toggle="collapse" data-bs-target="#FlexSearchCollapse" aria-expanded="false" aria-controls="FlexSearchCollapse">
                        <span class="material-icons size-20 menu-icon align-middle">search</span>
                        <span class="flexsearch-button-placeholder ms-1 me-2 d-none d-sm-block">Search</span>
                        <div class="d-none d-sm-block">
                            <span class="flexsearch-button-keys">
                                <kbd class="flexsearch-button-cmd-key">
                                    <svg width="44" height="15"><path d="M2.118,11.5A1.519,1.519,0,0,1,1,11.042,1.583,1.583,0,0,1,1,8.815a1.519,1.519,0,0,1,1.113-.458h.715V6.643H2.118A1.519,1.519,0,0,1,1,6.185,1.519,1.519,0,0,1,.547,5.071,1.519,1.519,0,0,1,1,3.958,1.519,1.519,0,0,1,2.118,3.5a1.519,1.519,0,0,1,1.114.458A1.519,1.519,0,0,1,3.69,5.071v.715H5.4V5.071A1.564,1.564,0,0,1,6.976,3.5,1.564,1.564,0,0,1,8.547,5.071,1.564,1.564,0,0,1,6.976,6.643H6.261V8.357h.715a1.575,1.575,0,0,1,1.113,2.685,1.583,1.583,0,0,1-2.227,0A1.519,1.519,0,0,1,5.4,9.929V9.214H3.69v.715a1.519,1.519,0,0,1-.458,1.113A1.519,1.519,0,0,1,2.118,11.5Zm0-.857a.714.714,0,0,0,.715-.714V9.214H2.118a.715.715,0,1,0,0,1.429Zm4.858,0a.715.715,0,1,0,0-1.429H6.261v.715a.714.714,0,0,0,.715.714ZM3.69,8.357H5.4V6.643H3.69ZM2.118,5.786h.715V5.071a.714.714,0,0,0-.715-.714.715.715,0,0,0-.5,1.22A.686.686,0,0,0,2.118,5.786Zm4.143,0h.715a.715.715,0,0,0,.5-1.22.715.715,0,0,0-1.22.5Z" fill="currentColor"></path><path d="M12.4,11.475H11.344l3.879-7.95h1.056Z" fill="currentColor"></path><path d="M25.073,5.384l-.864.576a2.121,2.121,0,0,0-1.786-.923,2.207,2.207,0,0,0-2.266,2.326,2.206,2.206,0,0,0,2.266,2.325,2.1,2.1,0,0,0,1.782-.918l.84.617a3.108,3.108,0,0,1-2.622,1.293,3.217,3.217,0,0,1-3.349-3.317,3.217,3.217,0,0,1,3.349-3.317A3.046,3.046,0,0,1,25.073,5.384Z" fill="currentColor"></path><path d="M30.993,5.142h-2.07v5.419H27.891V5.142h-2.07V4.164h5.172Z" fill="currentColor"></path><path d="M34.67,4.164c1.471,0,2.266.658,2.266,1.851,0,1.087-.832,1.809-2.134,1.855l2.107,2.691h-1.28L33.591,7.87H33.07v2.691H32.038v-6.4Zm-1.6.969v1.8h1.572c.832,0,1.22-.3,1.22-.918s-.411-.882-1.22-.882Z" fill="currentColor"></path><path d="M42.883,10.561H38.31v-6.4h1.033V9.583h3.54Z" fill="currentColor"></path></svg>
                                </kbd>
                                <kbd class="flexsearch-button-key">
                                    <svg width="15" height="15"><path d="M5.926,12.279H4.41L9.073,2.721H10.59Z" fill="currentColor"/></svg>
                                </kbd>
                            </span>
                        </div>
                    </button>
                
            </div>

        <div class="d-flex align-items-center">
            <ul class="list-unstyled mb-0">
                
            </ul>
            
        </div>
    </div>
    
    
            <div class="collapse" id="FlexSearchCollapse">
                <div class="flexsearch-container">
                    <div class="flexsearch-keymap">
                        <li>
                            <kbd class="flexsearch-button-cmd-key"><svg width="15" height="15" aria-label="Arrow down" role="img"><g fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.2"><path d="M7.5 3.5v8M10.5 8.5l-3 3-3-3"></path></g></svg></kbd>
                            <kbd class="flexsearch-button-cmd-key"><svg width="15" height="15" aria-label="Arrow up" role="img"><g fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.2"><path d="M7.5 11.5v-8M10.5 6.5l-3-3-3 3"></path></g></svg></kbd>
                            <span class="flexsearch-key-label">to navigate</span>
                        </li>
                        <li>
                            <kbd class="flexsearch-button-cmd-key"><svg width="15" height="15" aria-label="Enter key" role="img"><g fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.2"><path d="M12 3.53088v3c0 1-1 2-2 2H4M7 11.53088l-3-3 3-3"></path></g></svg></kbd>
                            <span class="flexsearch-key-label">to select</span>
                        </li>
                        <li>
                            <kbd class="flexsearch-button-cmd-key"><svg width="15" height="15" aria-label="Escape key" role="img"><g fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.2"><path d="M13.6167 8.936c-.1065.3583-.6883.962-1.4875.962-.7993 0-1.653-.9165-1.653-2.1258v-.5678c0-1.2548.7896-2.1016 1.653-2.1016.8634 0 1.3601.4778 1.4875 1.0724M9 6c-.1352-.4735-.7506-.9219-1.46-.8972-.7092.0246-1.344.57-1.344 1.2166s.4198.8812 1.3445.9805C8.465 7.3992 8.968 7.9337 9 8.5c.032.5663-.454 1.398-1.4595 1.398C6.6593 9.898 6 9 5.963 8.4851m-1.4748.5368c-.2635.5941-.8099.876-1.5443.876s-1.7073-.6248-1.7073-2.204v-.4603c0-1.0416.721-2.131 1.7073-2.131.9864 0 1.6425 1.031 1.5443 2.2492h-2.956"></path></g></svg></kbd>
                            <span class="flexsearch-key-label">to close</span>
                        </li>
                    </div>
                    <form class="flexsearch position-relative flex-grow-1 ms-2 me-2">
                        <div class="d-flex flex-row">
                            <input id="flexsearch" class="form-control" type="search" placeholder="Search" aria-label="Search" autocomplete="off">
                            <button id="hideFlexsearch" type="button" class="ms-2 btn btn-soft">
                                cancel
                            </button>
                        </div>
                        <div id="suggestions" class="shadow rounded-1 d-none"></div>
                    </form>
                </div>
            </div>
        
    
    
</div>


                            <div class="container-fluid">
                                <div class="layout-spacing">
                                    
                                        <div class="d-md-flex justify-content-between align-items-center"><nav aria-label="breadcrumb" class="d-inline-block pb-2 mt-1 mt-sm-0">
    <ul id="breadcrumbs" class="breadcrumb bg-transparent mb-0" itemscope itemtype="https://schema.org/BreadcrumbList">
        
            
                <li class="breadcrumb-item text-capitalize active" aria-current="page" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                    <a itemprop="item" href="/bdi-viz-manual/docs/">
                        <i class="material-icons size-20 align-text-bottom" itemprop="name">Home</i>
                    </a>
                    <meta itemprop="position" content='1' />
                </li>
            
            
                <li class="breadcrumb-item text-capitalize" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                    <a itemprop="item" href="/bdi-viz-manual/docs/get-started/">
                        <span itemprop="name">Get Started</span>
                    </a>
                    <meta itemprop="position" content='2' />
                </li>
            
        
            <li class="breadcrumb-item text-capitalize active" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
                <span itemprop="name">Dataset to GDC Schema Alignment</span>
                <meta itemprop="position" content='3' />
            </li>
        
    </ul>
</nav></div>
                                    
                                    <div class="row flex-xl-nowrap">
                                        
                                        <div class="docs-toc col-xl-3    d-xl-block"><toc>
    <div class="fw-bold text-uppercase mb-2">On this page</div>
    <nav id="TableOfContents">
  <ul>
    <li><a href="#installation-">Installation üõ†Ô∏è</a></li>
    <li><a href="#introduction-">Introduction üìö</a></li>
    <li><a href="#step-1-dataset-preparation-and-task-initialization-">Step 1: Dataset Preparation and Task Initialization üöÄ</a></li>
    <li><a href="#step-2-validating-high-confidence-matches-">Step 2: Validating High-Confidence Matches ‚úÖ</a></li>
    <li><a href="#step-3-explore-complex-attribute-matches-">Step 3: Explore Complex Attribute Matches üß©</a></li>
    <li><a href="#step-4-leverage-the-llm-agent-for-complex-matching-decisions-">Step 4: Leverage the LLM Agent for Complex Matching Decisions ü§ñ</a></li>
    <li><a href="#step-5-systematic-review-of-all-source-attributes-">Step 5: Systematic Review of All Source Attributes üîç</a></li>
    <li><a href="#step-6-exporting-the-gdc-compliant-dataset-">Step 6: Exporting the GDC-Compliant Dataset üì§</a></li>
    <li><a href="#further-resources-">Further Resources üìñ</a></li>
  </ul>
</nav>
    </toc></div>
                                        
                                        
                                        <div class="docs-toc-mobile    d-print-none d-xl-none">
                                            <button id="toc-dropdown-btn" class="btn-secondary dropdown-toggle" type="button" data-bs-toggle="dropdown" data-bs-offset="0,0" aria-expanded="false">
                                                Table of Contents
                                            </button>
<nav id="toc-mobile">
  <ul class="dropdown-menu">
    <li><a href="#installation-">Installation üõ†Ô∏è</a></li>
    <li><a href="#introduction-">Introduction üìö</a></li>
    <li><a href="#step-1-dataset-preparation-and-task-initialization-">Step 1: Dataset Preparation and Task Initialization üöÄ</a></li>
    <li><a href="#step-2-validating-high-confidence-matches-">Step 2: Validating High-Confidence Matches ‚úÖ</a></li>
    <li><a href="#step-3-explore-complex-attribute-matches-">Step 3: Explore Complex Attribute Matches üß©</a></li>
    <li><a href="#step-4-leverage-the-llm-agent-for-complex-matching-decisions-">Step 4: Leverage the LLM Agent for Complex Matching Decisions ü§ñ</a></li>
    <li><a href="#step-5-systematic-review-of-all-source-attributes-">Step 5: Systematic Review of All Source Attributes üîç</a></li>
    <li><a href="#step-6-exporting-the-gdc-compliant-dataset-">Step 6: Exporting the GDC-Compliant Dataset üì§</a></li>
    <li><a href="#further-resources-">Further Resources üìñ</a></li>
  </ul>
</nav></div>
                                        <div class="docs-content col-12 col-xl-9 mt-0">
                                            <div class="mb-0 d-flex">
                                                
                                                <h1 class="content-title mb-0">
                                                    Dataset to GDC Schema Alignment
                                                    
                                                </h1>
                                            </div>
                                            
                                                <p class="lead mb-3">Comprehensive guide for mapping biomedical datasets to the GDC schema standard</p>
                                            
                                            <div id="content" class="main-content" data-bs-spy="scroll" data-bs-root-margin="0px 0px -65%" data-bs-target="#toc-mobile">
                                                
    
    <div data-prismjs-copy="" data-prismjs-copy-success="" data-prismjs-copy-error="">
        <h2 id="installation-">Installation üõ†Ô∏è <a href="#installation-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>For system-specific installation instructions:</p>
<ul>
<li>AMD64 architecture: Follow the <a href="/bdi-viz-manual/docs/installation/install-on-linux/">Linux/Unix installation guide</a></li>
<li>ARM64 architecture (Apple Silicon): Follow the <a href="/bdi-viz-manual/docs/installation/install-on-macos/">MacOS installation guide</a></li>
</ul>
<h2 id="introduction-">Introduction üìö <a href="#introduction-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>This guide demonstrates the process of aligning a biomedical dataset with the Genomic Data Commons (GDC) schema using BDIViz. The GDC maintains standardized data repositories that require specific formatting and schema compliance for successful data submission and integration.</p>
<h2 id="step-1-dataset-preparation-and-task-initialization-">Step 1: Dataset Preparation and Task Initialization üöÄ <a href="#step-1-dataset-preparation-and-task-initialization-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><ol>
<li>Download the <a href="https://drive.google.com/file/d/1N3rbTHtnVDe19kMNei0opy_g-8Hr_Hl5/view?usp=drive_link" rel="external" target="_blank">Clark et al. sample dataset<svg width="16" height="16" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path fill="currentColor" d="M14 5c-.552 0-1-.448-1-1s.448-1 1-1h6c.552 0 1 .448 1 1v6c0 .552-.448 1-1 1s-1-.448-1-1v-3.586l-7.293 7.293c-.391.39-1.024.39-1.414 0-.391-.391-.391-1.024 0-1.414l7.293-7.293h-3.586zm-9 2c-.552 0-1 .448-1 1v11c0 .552.448 1 1 1h11c.552 0 1-.448 1-1v-4.563c0-.552.448-1 1-1s1 .448 1 1v4.563c0 1.657-1.343 3-3 3h-11c-1.657 0-3-1.343-3-3v-11c0-1.657 1.343-3 3-3h4.563c.552 0 1 .448 1 1s-.448 1-1 1h-4.563z"/></svg></a></li>
<li>Navigate to <code>http://localhost:3000</code> in your web browser</li>
<li>Select <strong>&ldquo;New Matching Task&rdquo;</strong> from the navigation bar</li>
<li>Upload the Clark dataset CSV file to the <strong>Source CSV File</strong> drop zone</li>
<li>Leave the <strong>Target CSV File</strong> and <strong>Target Schema JSON File</strong> fields empty to utilize the default GDC v3.3.0 schema</li>
<li>Click <strong>&ldquo;Import CSV&rdquo;</strong> to initiate the matching process</li>
</ol>
<p>





  



  
    
      <img src="/bdi-viz-manual/images/new-matching-task.gif" alt="new-matching-task-gif" width="1143" height="607" loading="lazy">
    
  
</p>
<h2 id="step-2-validating-high-confidence-matches-">Step 2: Validating High-Confidence Matches ‚úÖ <a href="#step-2-validating-high-confidence-matches-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>The system automatically identifies and suggests high-confidence matches, indicated by darker cells in the heatmap. Review these initial suggestions:</p>
<ol>
<li>
<p><strong>Direct Attribute Matches</strong>:</p>
<ul>
<li>Locate the matching candidate <strong>Gender</strong> (source) -&gt; <strong>gender</strong> (GDC) in the heatmap cell</li>
<li>Click the heatmap cell to expand the embedded node</li>
<li>The embedded node contains unique value distributions for both attributes</li>
<li>Verify that both attribute names and values align (Male ‚Üí Male, Female ‚Üí Female)</li>
<li><strong>Note that high-confidence matches are pre-accepted (indicated by green cells with checkmarks)</strong></li>
</ul>
<p>





  



  
    
      <img src="/bdi-viz-manual/images/clark-gender.gif" alt="clark-gender-gif" width="1108" height="666" loading="lazy">
    
  
</p>
</li>
<li>
<p><strong>Numerical Attribute Alignment</strong>:</p>
<ul>
<li>Examine the matching candidate <strong>BMI</strong> (source) -&gt; <strong>bmi</strong> (GDC) in the heatmap cell</li>
<li>Click the heatmap cell to expand the embedded node</li>
<li>The embedded node shows value bin distributions for numerical data</li>
<li>Observe that both attribute names and numerical distributions correspond appropriately</li>
<li><strong>This match should also be pre-accepted (green cell with checkmark)</strong></li>
</ul>
<p>





  



  
    
      <img src="/bdi-viz-manual/images/clark-bmi.gif" alt="clark-bmi-gif" width="1108" height="666" loading="lazy">
    
  
</p>
</li>
</ol>
<h2 id="step-3-explore-complex-attribute-matches-">Step 3: Explore Complex Attribute Matches üß© <a href="#step-3-explore-complex-attribute-matches-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>Address more nuanced matching scenarios that require human judgment:</p>
<ol>
<li>
<p><strong>Evaluating Semantic Differences</strong>:</p>
<ul>
<li>Locate the matching candidate <strong>Ethnicity_Self_Identify</strong> (source) -&gt; <strong>ethnicity</strong> (GDC) in the heatmap cell</li>
<li>Click the heatmap cell to expand the embedded node</li>
<li>Scroll down to check the value comparisons table in the embedded node</li>
<li>Note the significant value discrepancies:
<ul>
<li>Source values include &ldquo;Asian&rdquo;, &ldquo;White&rdquo;, etc.</li>
<li>GDC ethnicity values include &ldquo;hispanic or latino&rdquo;, &ldquo;not hispanic or latino&rdquo;, etc.</li>
</ul>
</li>
<li>Click the <strong>Reject</strong> (‚úï) button on the top left Shortcut Panel to invalidate this incorrect match</li>
</ul>
</li>
<li>
<p><strong>Identifying Cross-Category Matches</strong>:</p>
<ul>
<li>Examine the matching candidate <strong>Ethnicity_Self_Identify</strong> (source) -&gt; <strong>race</strong> (GDC) in the heatmap cell</li>
<li>Click the heatmap cell to expand the embedded node</li>
<li>Scroll down to check the value comparisons table in the embedded node</li>
<li>Confirm that the values correspond appropriately:
<ul>
<li>&ldquo;Asian&rdquo; in source aligns with &ldquo;asian&rdquo; in GDC</li>
<li>&ldquo;White&rdquo; in source aligns with &ldquo;white&rdquo; in GDC</li>
</ul>
</li>
<li>Click the <strong>Accept</strong> (‚úì) button on the top left Shortcut Panel to confirm this cross-category match</li>
</ul>
<p>





  



  
    
      <img src="/bdi-viz-manual/images/clark-ethnicity.gif" alt="clark-ethnicity-gif" width="1108" height="666" loading="lazy">
    
  
</p>
</li>
</ol>
<h2 id="step-4-leverage-the-llm-agent-for-complex-matching-decisions-">Step 4: Leverage the LLM Agent for Complex Matching Decisions ü§ñ <a href="#step-4-leverage-the-llm-agent-for-complex-matching-decisions-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>The integrated LLM Agent provides valuable insights for challenging attribute matches:</p>
<ol>
<li>
<p><strong>Evaluating Pathological Staging to Lymph Node Count</strong>:</p>
<ul>
<li>Examine the potential match <strong>Path_Stage_Reg_Lymph_Nodes_pN</strong> (source) -&gt; <strong>lymph_nodes_positive</strong> (GDC)</li>
<li>Consult the Agent Panel on the left side of the interface</li>
<li>Review the &ldquo;Semantic No Match&rdquo; card which indicates:
<ul>
<li>Source attribute represents a classification system (pN0, pN1, etc.)</li>
<li>Target attribute represents a numerical count of positive lymph nodes</li>
</ul>
</li>
<li>Based on this analysis, reject this match using the Shortcut Panel</li>
</ul>
</li>
<li>
<p><strong>Distinguishing Staging from Testing Metrics</strong>:</p>
<ul>
<li>Evaluate the potential match <strong>Path_Stage_Reg_Lymph_Nodes_pN</strong> (source) -&gt; <strong>lymph_nodes_tested</strong> (GDC)</li>
<li>Expand the &ldquo;Pattern No Match&rdquo; card in the Agent Panel</li>
<li>Note the critical insight that source values (pN0, pN1, pNX) represent staging categories</li>
<li>Confirm these do not align with the expected numeric values for lymph nodes tested</li>
<li>Reject this match using the Shortcut Panel</li>
</ul>
</li>
<li>
<p><strong>Identifying Non-Obvious Semantic Equivalence</strong>:</p>
<ul>
<li>Assess the potential match <strong>Path_Stage_Reg_Lymph_Nodes_pN</strong> (source) -&gt; <strong>ajcc_pathologic_n</strong> (GDC)</li>
<li>Despite different naming conventions, examine the value distributions</li>
<li>Consult the &ldquo;Semantic Match&rdquo; card in the Agent Panel</li>
<li>Verify that both attributes represent pathological lymph node staging in cancer systems</li>
<li>Accept this match using the Shortcut Panel</li>
</ul>
</li>
</ol>
<p>





  



  
    
      <img src="/bdi-viz-manual/images/clark-pn.gif" alt="clark-pn-gif" width="1418" height="666" loading="lazy">
    
  
</p>
<h2 id="step-5-systematic-review-of-all-source-attributes-">Step 5: Systematic Review of All Source Attributes üîç <a href="#step-5-systematic-review-of-all-source-attributes-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>Methodically evaluate each remaining source attribute:</p>
<ol>
<li>Prioritize heatmap cells with darker shading (indicating stronger match candidates)</li>
<li>Click each heatmap cell to expand the embedded node and evaluate both attribute names and value distributions</li>
<li>Utilize the top left Shortcut Panel to:
<ul>
<li><strong>Accept</strong> (‚úì) valid and appropriate matches</li>
<li><strong>Reject</strong> (‚úï) invalid or misaligned matches</li>
<li><strong>Discard</strong> (üóëÔ∏è) attributes that should be excluded from the mapping process</li>
</ul>
</li>
</ol>
<h2 id="step-6-exporting-the-gdc-compliant-dataset-">Step 6: Exporting the GDC-Compliant Dataset üì§ <a href="#step-6-exporting-the-gdc-compliant-dataset-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>After completing the evaluation process:</p>
<ol>
<li>Select the <strong>Export</strong> button from the top left Shortcut Panel</li>
<li>Choose CSV format to generate a dataset with GDC-compliant column names</li>
<li>Save the transformed file to your local system</li>
</ol>
<p>





  



  
    
      <img src="/bdi-viz-manual/images/export-matches.gif" alt="export-matches-gif" width="1530" height="690" loading="lazy">
    
  
</p>
<h2 id="further-resources-">Further Resources üìñ <a href="#further-resources-" class="anchor" aria-hidden="true"><i class="material-icons align-middle">link</i></a></h2><p>For advanced functionality and detailed operational guidance, please refer to the comprehensive <a href="/bdi-viz-manual/docs/manual/">User Manual</a>.</p>

    </div>

    

    
                                            </div>
                                            <div><hr class="doc-hr">
<div id="doc-nav" class="d-print-none">

	<div class="row flex-xl-nowrap ">
	<div class="col-sm-6 pt-2 doc-next">
		<a href="/bdi-viz-manual/docs/manual/upload-your-data/">
			<div class="card h-100 my-1">
				<div class="card-body py-2">
                    <p class="card-title fs-5 fw-semibold lh-base mb-0"><i class="material-icons align-middle">navigate_before</i> Upload Your Data</p>
					<p class="card-text ms-2">Begin a new schema matching ‚Ä¶</p>
					
				</div>
			</div>
		</a>
        </div>
	<div class="col-sm-6 pt-2 doc-prev">
		<a class="ms-auto" href="/bdi-viz-manual/docs/manual/explore-matches/">
			<div class="card h-100 my-1 text-end">
				<div class="card-body py-2">
                    <p class="card-title fs-5 fw-semibold lh-base mb-0">Explore Matches <i class="material-icons align-middle">navigate_next</i></p>
					<p class="card-text me-2">Explore schema alignment ‚Ä¶</p>
					
				</div>
			</div>
		</a>
        </div>
	</div>
</div></div>
                                        </div>
                                    </div>
                                </div>
                            </div>
<footer class="shadow py-3 d-print-none">
    <div class="container-fluid">
        <div class="row align-items-center">
            <div class="col">
                <div class="text-sm-start text-center mx-md-2">
                    <p class="mb-0">
                        
                        ¬© 2025 NYU VIDA Lab.
                    </p>
                </div>
            </div>
        </div>
    </div>
</footer>
</main>
            </div>
        </div>

        
        
        <button onclick="topFunction()" id="back-to-top" aria-label="Back to Top Button" class="back-to-top fs-5"><svg width="24" height="24"><path d="M12,10.224l-6.3,6.3L4.32,15.152,12,7.472l7.68,7.68L18.3,16.528Z" style="fill:#fff"/></svg></button>
        
        

        
        




    
    
    






    <script src="/bdi-viz-manual/docs/js/bootstrap.js" defer></script>


    <script type="text/javascript" src="http://localhost:1313/bdi-viz-manual/docs/js/bundle.js" defer></script>

        

        
        <script type="module">
    var suggestions = document.getElementById('suggestions');
    var search = document.getElementById('flexsearch');

    const flexsearchContainer = document.getElementById('FlexSearchCollapse');

    const hideFlexsearchBtn = document.getElementById('hideFlexsearch');

    const configObject = { toggle: false }
    const flexsearchContainerCollapse = new Collapse(flexsearchContainer, configObject) 

    if (search !== null) {
        document.addEventListener('keydown', inputFocus);
        flexsearchContainer.addEventListener('shown.bs.collapse', function () {
            search.focus();
        });
        
        var topHeader = document.getElementById("top-header");
        document.addEventListener('click', function(elem) {
            if (!flexsearchContainer.contains(elem.target) && !topHeader.contains(elem.target))
                flexsearchContainerCollapse.hide();
        });
    }

    hideFlexsearchBtn.addEventListener('click', () =>{
        flexsearchContainerCollapse.hide()
    })

    function inputFocus(e) {
        if (e.ctrlKey && e.key === '/') {
            e.preventDefault();
            flexsearchContainerCollapse.toggle();
        }
        if (e.key === 'Escape' ) {
            search.blur();
            
            flexsearchContainerCollapse.hide();
        }
    };

    document.addEventListener('click', function(event) {

    var isClickInsideElement = suggestions.contains(event.target);

    if (!isClickInsideElement) {
        suggestions.classList.add('d-none');
    }

    });

    


    document.addEventListener('keydown',suggestionFocus);

    function suggestionFocus(e) {
    const suggestionsHidden = suggestions.classList.contains('d-none');
    if (suggestionsHidden) return;

    const focusableSuggestions= [...suggestions.querySelectorAll('a')];
    if (focusableSuggestions.length === 0) return;

    const index = focusableSuggestions.indexOf(document.activeElement);

    if (e.key === "ArrowUp") {
        e.preventDefault();
        const nextIndex = index > 0 ? index - 1 : 0;
        focusableSuggestions[nextIndex].focus();
    }
    else if (e.key === "ArrowDown") {
        e.preventDefault();
        const nextIndex= index + 1 < focusableSuggestions.length ? index + 1 : index;
        focusableSuggestions[nextIndex].focus();
    }

    }

    


    (function(){

    var index = new FlexSearch.Document({
        
        tokenize: "forward",
        minlength:  0 ,
        cache:  100 ,
        optimize:  true ,
        document: {
        id: 'id',
        store: [
            "href", "title", "description"
        ],
        index: ["title", "description", "content"]
        }
    });


    


    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    


    

    

    index.add(
            {
                id:  0 ,
                href: "\/bdi-viz-manual\/docs\/manual\/upload-your-data\/",
                title: "Upload Your Data",
                description: "Begin a new schema matching task by uploading your dataset files.",
                content: "1. Start a New Matching Task üîÑ linkTo begin, click the ‚ÄúNew Matching Task‚Äù button located on the far right of the navigation bar. This will open the upload interface.\nThe interface presents three drop zones:\nSource CSV File ‚Äì Your input dataset Target CSV File ‚Äì The standardized reference dataset Target Schema JSON File ‚Äì Metadata for interpreting the reference schema These files are essential for initiating schema matching and visual comparison.\n2. Upload Required Files üìÇ linkA. Source CSV File (Required) linkUpload a .csv file containing your unstructured or raw biomedical dataset. This will be used as the basis for matching.\nExample:\nCase_ID Gender Ethnicity_Self_Identify Path_Stage_Reg_Lymph_Nodes_pN C1234 Male Asian pN1 C5678 Female White pN2 C1111 Unknown Unknown pNX B. Target CSV File (Optional) linkIf omitted, the system will default to the GDC v3.3.0 schema.\nUpload a .csv file that reflects the reference or gold-standard biomedical dataset (e.g., GDC, PDC).\nExample:\nsubmitter_id gender race ajcc_pathologic_n T001 Male Asian N1 T002 Female White N2 C. Target Schema JSON File (Optional) linkIf omitted, the system will use the GDC v3.3.0 schema definition.\nUpload a .json file that defines metadata and attributes for the target schema.\nExample:\n{ \"age_at_index\": { \"column_name\": \"age_at_index\", \"category\": \"clinical\", \"node\": \"demographic\", \"type\": \"integer\", \"description\": \"The patient's age (in years) on the reference or anchor date used during date obfuscation.\", \"minimum\": 0 }, \"education_level\": { \"column_name\": \"education_level\", \"category\": \"clinical\", \"node\": \"demographic\", \"type\": \"enum\", \"description\": \"The years of schooling completed in graded public, private, or parochial schools, and in colleges, universities, or professional schools.\", \"enum\": [ \"College Degree\", \"High School Graduate or GED\", \"Professional or Graduate Degree\", \"Some High School or Less\", \"Vocational College or Some College\", \"Unknown\", \"Not Reported\" ] } } 3. Launch the Matching Task üöÄ linkOnce all necessary files are uploaded, click the ‚ÄúImport CSV‚Äù button to start processing.\nThe system will:\nAnalyze your source data in relation to the reference schema Generate schema alignment recommendations Display a loading indicator while processing is in progress This step may take several minutes depending on the size of your dataset.\nWhat‚Äôs Next? üöÄ linkAfter the task is launched, you will be redirected to the schema matching workspace where you can:\nReview alignment suggestions Accept or reject proposed matches Explore value mappings and distributions interactively "
            }
        );
    index.add(
            {
                id:  1 ,
                href: "\/bdi-viz-manual\/docs\/get-started\/dataset-to-gdc\/",
                title: "Dataset to GDC Schema Alignment",
                description: "Comprehensive guide for mapping biomedical datasets to the GDC schema standard",
                content: "Installation üõ†Ô∏è linkFor system-specific installation instructions:\nAMD64 architecture: Follow the Linux/Unix installation guide ARM64 architecture (Apple Silicon): Follow the MacOS installation guide Introduction üìö linkThis guide demonstrates the process of aligning a biomedical dataset with the Genomic Data Commons (GDC) schema using BDIViz. The GDC maintains standardized data repositories that require specific formatting and schema compliance for successful data submission and integration.\nStep 1: Dataset Preparation and Task Initialization üöÄ link Download the Clark et al. sample dataset Navigate to http://localhost:3000 in your web browser Select ‚ÄúNew Matching Task‚Äù from the navigation bar Upload the Clark dataset CSV file to the Source CSV File drop zone Leave the Target CSV File and Target Schema JSON File fields empty to utilize the default GDC v3.3.0 schema Click ‚ÄúImport CSV‚Äù to initiate the matching process Step 2: Validating High-Confidence Matches ‚úÖ linkThe system automatically identifies and suggests high-confidence matches, indicated by darker cells in the heatmap. Review these initial suggestions:\nDirect Attribute Matches:\nLocate the matching candidate Gender (source) -\u003e gender (GDC) in the heatmap cell Click the heatmap cell to expand the embedded node The embedded node contains unique value distributions for both attributes Verify that both attribute names and values align (Male ‚Üí Male, Female ‚Üí Female) Note that high-confidence matches are pre-accepted (indicated by green cells with checkmarks) Numerical Attribute Alignment:\nExamine the matching candidate BMI (source) -\u003e bmi (GDC) in the heatmap cell Click the heatmap cell to expand the embedded node The embedded node shows value bin distributions for numerical data Observe that both attribute names and numerical distributions correspond appropriately This match should also be pre-accepted (green cell with checkmark) Step 3: Explore Complex Attribute Matches üß© linkAddress more nuanced matching scenarios that require human judgment:\nEvaluating Semantic Differences:\nLocate the matching candidate Ethnicity_Self_Identify (source) -\u003e ethnicity (GDC) in the heatmap cell Click the heatmap cell to expand the embedded node Scroll down to check the value comparisons table in the embedded node Note the significant value discrepancies: Source values include ‚ÄúAsian‚Äù, ‚ÄúWhite‚Äù, etc. GDC ethnicity values include ‚Äúhispanic or latino‚Äù, ‚Äúnot hispanic or latino‚Äù, etc. Click the Reject (‚úï) button on the top left Shortcut Panel to invalidate this incorrect match Identifying Cross-Category Matches:\nExamine the matching candidate Ethnicity_Self_Identify (source) -\u003e race (GDC) in the heatmap cell Click the heatmap cell to expand the embedded node Scroll down to check the value comparisons table in the embedded node Confirm that the values correspond appropriately: ‚ÄúAsian‚Äù in source aligns with ‚Äúasian‚Äù in GDC ‚ÄúWhite‚Äù in source aligns with ‚Äúwhite‚Äù in GDC Click the Accept (‚úì) button on the top left Shortcut Panel to confirm this cross-category match Step 4: Leverage the LLM Agent for Complex Matching Decisions ü§ñ linkThe integrated LLM Agent provides valuable insights for challenging attribute matches:\nEvaluating Pathological Staging to Lymph Node Count:\nExamine the potential match Path_Stage_Reg_Lymph_Nodes_pN (source) -\u003e lymph_nodes_positive (GDC) Consult the Agent Panel on the left side of the interface Review the ‚ÄúSemantic No Match‚Äù card which indicates: Source attribute represents a classification system (pN0, pN1, etc.) Target attribute represents a numerical count of positive lymph nodes Based on this analysis, reject this match using the Shortcut Panel Distinguishing Staging from Testing Metrics:\nEvaluate the potential match Path_Stage_Reg_Lymph_Nodes_pN (source) -\u003e lymph_nodes_tested (GDC) Expand the ‚ÄúPattern No Match‚Äù card in the Agent Panel Note the critical insight that source values (pN0, pN1, pNX) represent staging categories Confirm these do not align with the expected numeric values for lymph nodes tested Reject this match using the Shortcut Panel Identifying Non-Obvious Semantic Equivalence:\nAssess the potential match Path_Stage_Reg_Lymph_Nodes_pN (source) -\u003e ajcc_pathologic_n (GDC) Despite different naming conventions, examine the value distributions Consult the ‚ÄúSemantic Match‚Äù card in the Agent Panel Verify that both attributes represent pathological lymph node staging in cancer systems Accept this match using the Shortcut Panel Step 5: Systematic Review of All Source Attributes üîç linkMethodically evaluate each remaining source attribute:\nPrioritize heatmap cells with darker shading (indicating stronger match candidates) Click each heatmap cell to expand the embedded node and evaluate both attribute names and value distributions Utilize the top left Shortcut Panel to: Accept (‚úì) valid and appropriate matches Reject (‚úï) invalid or misaligned matches Discard (üóëÔ∏è) attributes that should be excluded from the mapping process Step 6: Exporting the GDC-Compliant Dataset üì§ linkAfter completing the evaluation process:\nSelect the Export button from the top left Shortcut Panel Choose CSV format to generate a dataset with GDC-compliant column names Save the transformed file to your local system Further Resources üìñ linkFor advanced functionality and detailed operational guidance, please refer to the comprehensive User Manual.\n"
            }
        );
    index.add(
            {
                id:  2 ,
                href: "\/bdi-viz-manual\/docs\/manual\/explore-matches\/",
                title: "Explore Matches",
                description: "Explore schema alignment candidates using the BDIViz interactive heatmap interface.",
                content: "Overview üîç linkAfter uploading your files, BDIViz launches its main interface, allowing you to visually explore match candidates between your source dataset and the target schema.\nInteractive Heatmap üî• linkThe heatmap is the primary visualization tool in BDIViz. Each cell represents a match candidate between a source attribute (y-axis) and a target attribute (x-axis).\nKey Features link Heatmap Cells X-Axis Hierarchy Expandable Nodes Heatmap Top Tabs Each cell corresponds to a potential match:\nColor intensity indicates similarity score (darker = stronger match). Green = accepted match, Red = rejected match. Click on a cell to inspect match details. The x-axis displays the target schema‚Äôs structure as a semantic hierarchy:\nCategory Level: e.g., clinical, biospecimen Node Level: e.g., diagnosis, treatment Leaf Nodes: individual target attributes Color and curved connectors help clarify relationships and improve navigation.\nClicking a heatmap node reveals a stacked histogram panel:\nTop Chart: Source column value distribution Bottom Chart: Target column value distribution Use this to evaluate whether the two columns share meaningful overlap.\nTip: Similar distributions (e.g., shared categories like ‚ÄúMale‚Äù and ‚ÄúFemale‚Äù) often suggest semantic alignment.\nThe top-level filter tabs help narrow your focus:\nAll: View all candidate matches Accepted: Only show confirmed matches Unmatched: Only show source columns with no confirmed match Expand on Hover: Toggle whether expanded histograms appear on hover or click Filters and Search Controls üîç linkFine-tune the heatmap view using a suite of filters:\nSource Attribute Selector Similarity Threshold Similar Attributes Displayed Search Bar Select a specific source column to examine.\nDropdown key:\nGreen: Already matched Grey: Manually discarded All: Show all source attributes Set a minimum similarity score for visible candidates. Range: 0.0 ‚Äì 1.0.\n0.0: Show all matches 1.0: Show only perfect matches Controls how many similar source attributes appear on the heatmap when a single source column is selected.\nNote: Only applies when Source Attribute ‚â† ‚ÄúAll‚Äù.\nQuickly locate and highlight target attributes by name or keyword.\nLower Panel: Match Details üîé linkClicking on any heatmap node reveals deeper insights below:\nValue Comparisons UpSet Plot Visualizes string-based fuzzy matching between source and target values.\nEach row: one source value + its closest matches Use to validate whether mapping is semantically and syntactically justified Visualizes how different matchers contributed to the candidate match score.\nEach row: a matcher Each column: a candidate Dots indicate support from that matcher What‚Äôs Next? üöÄ linkAfter reviewing the matches:\nAccept or reject individual match suggestions Use filtering to prioritize high-confidence matches Proceed to export, refine, or apply your matched schema for downstream harmonization tasks "
            }
        );
    index.add(
            {
                id:  3 ,
                href: "\/bdi-viz-manual\/docs\/installation\/install-on-linux\/",
                title: "Install on Linux/Unix",
                description: "Step-by-step guide for installing and configuring BDIViz on Linux and Unix-based systems",
                content: "Prerequisites linkBefore installing BDIViz on your Linux system, ensure you have the following prerequisites:\nA Linux distribution (Ubuntu, Debian, CentOS, etc.) with amd64 architecture Docker Engine installed and running (version 19.03 or later) At least 4GB of RAM and 10GB of free disk space Internet connection to pull the Docker image Install Docker (if not already installed) linkIf you don‚Äôt have Docker installed, follow these steps to install it:\nUbuntu/Debian CentOS/RHEL Arch Linux sudo apt update sudo apt install \\ ca-certificates \\ curl \\ gnupg \\ lsb-release sudo mkdir -p /etc/apt/keyrings curl -fsSL https://download.docker.com/linux/ubuntu/gpg | \\ sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg echo \\ \"deb [arch=$(dpkg --print-architecture) \\ signed-by=/etc/apt/keyrings/docker.gpg] \\ https://download.docker.com/linux/ubuntu \\ $(lsb_release -cs) stable\" | \\ sudo tee /etc/apt/sources.list.d/docker.list \u003e /dev/null sudo apt update sudo apt install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin sudo systemctl enable docker sudo systemctl start docker sudo yum install -y yum-utils sudo yum-config-manager \\ --add-repo https://download.docker.com/linux/centos/docker-ce.repo sudo yum install docker-ce docker-ce-cli containerd.io docker-compose-plugin sudo systemctl enable docker sudo systemctl start docker sudo pacman -Syu sudo pacman -S docker sudo systemctl enable docker sudo systemctl start docker Run BDIViz Using Docker linkOnce Docker is installed and running, follow these steps to run BDIViz using the edenwu/bdi-viz-react:amd64 image:\n1. Pull the Docker image link docker pull edenwu/bdi-viz-react:amd64 2. Run the container link docker run -d \\ --name bdiviz \\ -p 3000:3000 \\ edenwu/bdi-viz-react:amd64 This will start the BDIViz container in detached mode and expose the web application at http://localhost:3000.\n3. Verify the container is running linkYou can verify the container is up using:\ndocker ps If everything is working correctly, you should see bdiviz listed in the output.\n4. Stop the container linkWhen you‚Äôre done, you can stop the container with:\ndocker stop bdiviz And if needed, remove it:\ndocker rm bdiviz Note link Make sure port 3000 is available on your host machine. You can bind-mount local volumes or customize environment variables for advanced use (e.g., persistent storage, configuration injection). To update the image, simply pull the latest tag again and restart the container: docker pull edenwu/bdi-viz-react:amd64 docker stop bdiviz \u0026\u0026 docker rm bdiviz docker run -d --name bdiviz -p 3000:3000 edenwu/bdi-viz-react:amd64 Troubleshooting link Port already in use? Try using a different port: docker run -d --name bdiviz-mac -p 8080:3000 edenwu/bdi-viz-react:amd64 Then access it at http://localhost:8080. Docker daemon not running? Open Docker Desktop and make sure it‚Äôs active. Still stuck? Feel free to reach out or open an issue on the GitHub repository. "
            }
        );
    index.add(
            {
                id:  4 ,
                href: "\/bdi-viz-manual\/docs\/get-started\/verify-value-mappings\/",
                title: "Verify Value Matching",
                description: "Verify curated mappings from source attribute values to target attribute values",
                content: "Installation üõ†Ô∏è linkFor system-specific installation instructions (pick ONE for your CPU architecture):\nAMD64: follow the Linux/Unix installation guide ARM64 (Apple Silicon): follow the macOS installation guide Overview ‚úÖ linkThis guide walks you through verifying value mappings after schema alignment. You will enable Developer Mode, upload a ground truth CSV for value mapping verification, and use the heatmap and Data Wrangler to confirm results.\nStart a New Session üóÇÔ∏è link On the far left of the top navbar, enter a name in the New session field to create a new session. The new session inherits data from the default session, so you can proceed without re-uploading. Toggle Developer Mode üß∞ link How to open: Click the arrow icon on the top-left corner to open the side panel, then toggle Developer Mode. What it enables: Upload pre-annotated ground truth for schema matching (attribute pairs) Upload pre-annotated ground truth for value mappings (value-to-value) Upload custom matchers (not covered here) Developer Mode surfaces additional upload slots and debugging panels designed for power users and evaluators.\nPrepare Data and Start the Task üì¶ linkFor verifying value mappings, prepare the following:\nSource table (required): your raw dataset CSV, e.g., Cao.csv. Target table (not needed): leave empty to use the built-in GDC schema as target. Ground truth CSV (required for verification): a file with exactly four columns: source_attribute target_attribute source_value target_value Example (subset):\nsource_attribute,target_attribute,source_value,target_value FIGO_stage,figo_stage,IA,Stage IA FIGO_stage,figo_stage,IIIC1,Stage IIIC1 Ethnicity,ethnicity,Hispanic or Latino,hispanic or latino Race,race,White,white Gender,gender,Female,female After files are ready, click the blue Start Value Matching button at the bottom-right to launch the verification task.\nReview Schema Matches on the Heatmap üî• link The heatmap displays candidate matches between source (y-axis) and GDC target attributes (x-axis). Ground-truth schema pairs included in your CSV should appear as heatmap cells. Hover a cell to preview details; click to expand the embedded node with distributions and comparisons. Use the control panel filters to narrow candidates by similarity, attribute, or status. Need a refresher on filters and controls? See explore-matches.\nVerify in the Data Wrangler üìä linkWhen you click a heatmap node:\nThe lower DATA WRANGLER view focuses on the selected source attribute (highlighted with a blue background). The matched target attribute appears appended to the right of the source attribute. If a value mapping is provided in your ground truth, the target column shows the mapped target values applied to each source value. If no value mapping is provided for a source value, the corresponding cell in the target column will be empty. Use this view to scan for mismatches, missing mappings, or unexpected blanks. Adjust your ground truth and rerun if needed.\nEdit and Correct Mapped Values ‚úèÔ∏è linkIf a mapped value looks incorrect:\nHover the cell in the mapped target column within the DATA WRANGLER table. Choose one of the following: Inline edit: Click the current mapped value to edit directly, then press Enter to apply.\nEdit via popover: Click the edit icon on the right side of the cell to open a popover listing all available target schema values as selectable chips. Click a chip to apply it. Note: When you change the mapped value for a specific source_value, the update is applied to all rows in this source attribute that share the same source_value.\nUse the Harmonization Assistant for Value Mapping Suggestions linkUse the Harmonization Assistant to quickly suggest or apply value mappings for both categorical and numerical columns.\nSelect a heatmap node, then right-click to open the Harmonization Assistant panel. Ask the agent with a clear prompt: Example (categorical): ‚ÄúMap source values in Race to GDC race values.‚Äù Example (numerical): ‚ÄúSuggest bins for BMI aligned to GDC categories.‚Äù Review and apply the suggested mappings to update the target column. To reset the conversation, click the second button in the panel‚Äôs top-right to clear the chat history.\nSee the full assistant guide: /bdi-viz-manual/docs/manual/llm-agent/.\nExport Mapping Results üì§ linkAfter applying all changes, export your finalized value mappings:\nOpen the Shortcut Panel (top-left). Click Export Matching Results. In the format options, select CSV 4-column to generate a ground truth-like file. The exported CSV contains four columns:\nsource_attribute target_attribute source_value target_value "
            }
        );
    index.add(
            {
                id:  5 ,
                href: "\/bdi-viz-manual\/docs\/installation\/install-on-macos\/",
                title: "Install on MacOS",
                description: "Step-by-step guide for installing and configuring BDIViz on MacOS(ARM64) systems",
                content: "Prerequisites linkBefore setting up BDIViz on your Mac, ensure your system meets the following requirements:\nMac computer with Apple Silicon (M1, M2, M3, etc.) ‚Äì ARM64 architecture macOS 12.0 Monterey or later Docker Desktop for Mac (ARM64 version) At least 4GB of RAM and 10GB of free disk space Internet connection to pull the Docker image Install Docker Desktop for Mac link Download Docker Desktop for Mac (Apple Silicon): üëâ https://www.docker.com/products/docker-desktop/\nOpen the downloaded .dmg file and drag Docker to your Applications folder.\nLaunch Docker Desktop from Spotlight or Applications.\nGrant permissions if prompted and wait for Docker to start (you should see the Docker icon in the menu bar).\nOpen Terminal and verify installation:\ndocker --version You should see an output like:\nDocker version 24.0.0, build xxxxxxxx Run BDIViz Using Docker link1. Pull the Docker image link docker pull edenwu/bdi-viz-react:arm64 2. Run the container link docker run -d \\ --name bdiviz \\ -p 3000:3000 \\ edenwu/bdi-viz-react:arm64 This will start the BDIViz container in detached mode and expose the web application at http://localhost:3000.\n3. Verify the container is running linkYou can verify the container is up using:\ndocker ps If everything is working correctly, you should see bdiviz listed in the output.\n4. Stop the container linkWhen you‚Äôre done, you can stop the container with:\ndocker stop bdiviz And if needed, remove it:\ndocker rm bdiviz Note link Make sure port 3000 is available on your host machine. You can bind-mount local volumes or customize environment variables for advanced use (e.g., persistent storage, configuration injection). To update the image, simply pull the latest tag again and restart the container: docker pull edenwu/bdi-viz-react:arm64 docker stop bdiviz \u0026\u0026 docker rm bdiviz docker run -d --name bdiviz -p 3000:3000 edenwu/bdi-viz-react:arm64 Troubleshooting link Port already in use? Try using a different port: docker run -d --name bdiviz-mac -p 8080:3000 edenwu/bdi-viz-react:arm64 Then access it at http://localhost:8080. Docker daemon not running? Open Docker Desktop and make sure it‚Äôs active. Still stuck? Feel free to reach out or open an issue on the GitHub repository. "
            }
        );
    index.add(
            {
                id:  6 ,
                href: "\/bdi-viz-manual\/docs\/manual\/llm-agent\/",
                title: "LLM Agent",
                description: "Explore schema alignment candidates using the BDIViz LLM Agent.",
                content: "Overview üîç linkBDIViz features an embedded LLM-powered assistant to help you validate, explain, and refine schema alignment candidates. The agent provides contextual insights, interactive feedback, and harmonization support throughout your data mapping workflow.\nLLM Validation Panel ü§ñ link The LLM Validation Panel offers real-time, context-aware suggestions and explanations for each match candidate:\nDiagnosis Summary: Evaluates if a match is likely valid based on metadata, column names, and user history. Explanation Cards: Highlights key reasons for each suggestion (e.g., name similarity, value overlap). Click to expand for more detail. Provide feedback via üëç/üëé to refine model accuracy. Target Schema Metadata: Enriched descriptions pulled from sources such as GDC. Harmonization Assistant ü§ñ link The Harmonization Assistant acts as your copilot for schema mapping and harmonization tasks, offering interactive, conversational support.\nHow to Access link Toggle Top Search Bar Right-click Expanded Node Toggle the assistant button on the right side of the search bar. Right-click on a selected or expanded node. Example Q\u0026A Interactions linkWhat is this candidate about? linkA: Ask the assistant for a summary or description of the selected candidate.\nPrompt: What is this candidate about? What are the potential value mappings between the selected source and target attribute? linkA: Request the assistant to show possible value mappings between the source and target attributes.\nPrompt: What are the potential value mappings between the selected source and target attribute?\nDoes the candidate I selected make sense? Why? linkA: Ask the assistant to explain the reasoning behind a suggested match.\nPrompt: Does the candidate I selected make sense? Why?\nCan you suggest more candidates from the schema? linkA: Ask the assistant to search for additional candidates from the top recommendations or the entire target schema.\nPrompt: Besides the current recommendations, show me more candidates related to tumor sites or organs.\nCan you accept or reject candidates for me? linkA: Let the assistant accept or reject candidates based on chat history and its own suggestions.\nPrompt: From your memory and our chat, can you accept the candidate that makes the most sense?\nCan you remember custom notes or metadata for later? linkA: Add custom notes or metadata to the assistant‚Äôs memory for future reference.\nPrompt: Remember this metadata for the source dataset: CCRCC is a common problem among...\nCan you remind me what this task is about? linkA: Ask the assistant to recall the current task or context.\nPrompt: Recall from your memory, what is the context of this task?\nCan you filter or rematch nodes based on criteria? linkA: Start a new task or filter nodes based on clinical or semantic criteria.\nPrompt: Rematch with nodes related to clinical criteria.\nCan you reorder or rescore candidates? linkA: Request the assistant to reorder or rescore candidates.\nPrompt: Reorder the candidates based on relevance to AJCC. Rescore and update them. Can you help me create a custom matcher task? (unstable) linkA: Ask the assistant to help create new matcher tasks.\nPrompt: Help me add a matcher using the rapidfuzz library.\nTip: The Harmonization Assistant learns from your feedback and memory, improving its recommendations over time. Use the thumbs up/down and memory features to tailor the assistant to your project needs.\n"
            }
        );
    index.add(
            {
                id:  7 ,
                href: "\/bdi-viz-manual\/docs\/overview\/",
                title: "Overview",
                description: "BDIViz: An Interactive Visualization System for Biomedical Schema Matching with LLM-Supported Validation",
                content: "‚öôÔ∏è Getting Started linkFirst, install the required dependencies:\nnpm i . Then, run the development server:\nnpm run dev ‚öôÔ∏è Introduction linkBDIViz is an interactive web-based application developed as part of the ARPA-H ASKEM project to support schema matching and value mapping tasks in biomedical data integration. It provides users with a rich visual interface‚Äîincluding heatmaps, explanations, and value comparisons‚Äîto streamline the process of aligning raw biomedical datasets with standardized data schemas such as the Genomic Data Commons (GDC) and Proteomic Data Commons (PDC).\nBDIViz is model agnostic, meaning it can be used with any schema matching model. It is designed to work with the BDI-Kit module, which is a Python library that provides a set of tools for schema matching and value mapping tasks. The BDI-Kit module includes a variety of schema matching algorithms, including supervised and unsupervised methods, as well as tools for data preprocessing and feature extraction.\n‚öôÔ∏è Features link üîç Interactive Heatmap for exploring source-target column match candidates üìä Value Comparison Table using fuzzy matching on raw values ü§ñ LLM-Powered Agent Panel for dynamic match explanations and feedback ‚è™ Timeline View to trace user actions (accept, reject, discard) üéØ Control Panel for adjusting similarity threshold and navigating source columns üì§ Export Curated Mappings as JSON or CSV for downstream use üì∑ Demo linkVideo demo: Live Demo: https://bdiviz-dev.users.hsrn.nyu.edu/dashboard/\n"
            }
        );
    index.add(
            {
                id:  8 ,
                href: "\/bdi-viz-manual\/docs\/manual\/user-operation\/",
                title: "Perform User Operations",
                description: "Interactively evaluate and manage each schema match candidate.",
                content: "Overview üîç linkBDIViz provides a streamlined interface for managing match candidates directly within the heatmap view. The Shortcut Panel, located at the top-left of the screen, offers essential interactive tools for evaluating and adjusting alignment results.\nThese operations allow users to mark candidates as accepted, rejected, or discarded, while also providing undo/redo functionality for iterative refinement.\nAvailable Actions üéÆ linkThe table below summarizes the core user operations available via the Shortcut Panel:\nAction Icon Description Visual Indicator Accept ‚úì Confirms a source-to-target column pair as a valid schema match Cell turns green Reject ‚úï Flags the candidate as an invalid or incorrect match Cell turns red Discard üóëÔ∏è Removes the source attribute from matching consideration Attribute grays out Undo ‚ü≤ Reverts the most recent operation Previous state restored Redo ‚ü≥ Reapplies the most recently undone operation Last change re-applied All user operations are recorded in the Timeline Panel (located on the right), enabling traceability and decision auditing.\nBest Practices üí° link Leverage Keyboard Shortcuts: Speed up match decisions using shortcut buttons or hotkeys (where supported). Work Iteratively: Use Undo and Redo freely to experiment and refine decisions without permanent impact. Stay Organized: The Timeline Panel logs every action in sequence, helping you track progress or revisit past decisions. Interpret Colors Quickly: ‚úÖ Green: Accepted match ‚ùå Red: Rejected match üö´ Gray: Discarded source attribute What‚Äôs Next? üöÄ linkAfter reviewing and operating on match candidates:\nReview the Timeline Panel to validate key decisions. Use filters and the heatmap overview to identify remaining unmatched attributes. Proceed to export your confirmed mappings or refine further with the help of the LLM Agent Panel. "
            }
        );
    index.add(
            {
                id:  9 ,
                href: "\/bdi-viz-manual\/docs\/manual\/export-your-mappings\/",
                title: "Export Your Mappings",
                description: "Export the results of your schema matching task in multiple formats for downstream integration and analysis.",
                content: "Overview üìä linkOnce your schema matching task is complete, BDIViz enables you to export the validated mappings for use in data integration pipelines, reporting workflows, or system submissions.\nHow to Export üì§ linkTo begin exporting your results:\nNavigate to the Shortcut Panel (top-left corner of the interface). Click the Export button. A dialog will appear prompting you to select your preferred export format (CSV or JSON). Export Formats üìã linkBDIViz supports two export options designed for different use cases:\nCSV Format JSON Format The CSV export transforms your original dataset using only the accepted column mappings.\nKey Features: link Source columns are renamed to their matched target attributes Only accepted matches are included Resulting dataset is ready for immediate analysis or upload Example Output: link submitter_id gender race ajcc_pathologic_n C1234 Male Asian pN1 C5678 Female White pN2 C1111 Unknown Unknown pNX Best For: link Data harmonization pipelines Uploads to repositories (e.g., GDC, PDC) Sharing aligned datasets with collaborators The JSON export provides a structured record of all mapping operations with detailed metadata.\nKey Features: link Includes column-level mappings and value-level match pairs Records additional metadata like match scores and timestamps Machine-readable for downstream processing or reproducibility Example Output: link [ { \"sourceColumn\": \"Tumor_Focality\", \"targetColumn\": \"tumor_focality\", \"valueMatches\": [ { \"from\": \"Multifocal\", \"to\": \"Multifocal\" }, { \"from\": \"Unifocal\", \"to\": \"Unifocal\" } ] }, { \"sourceColumn\": \"Ethnicity\", \"targetColumn\": \"ethnicity\", \"valueMatches\": [ { \"from\": \"Hispanic or Latino\", \"to\": \"hispanic or latino\" }, { \"from\": \"Not reported\", \"to\": \"not reported\" }, { \"from\": \"Not-Hispanic or Latino\", \"to\": \"not hispanic or latino\" } ] } ] Best For: link Programmatic ingestion and audit trails Reuse in automated systems Reconstructing or refining schema alignment processes Final Notes üìù link Ensure all intended mappings are accepted before exporting; rejected and discarded mappings will not be included. Use the Timeline Panel to review decisions prior to export. Both formats can be downloaded directly to your local machine. "
            }
        );
    index.add(
            {
                id:  10 ,
                href: "\/bdi-viz-manual\/docs\/",
                title: "BDIViz Documentations",
                description: "Explore our guides and examples to deploy and improve your data integration workflow with BDIViz.",
                content: ""
            }
        );
    index.add(
            {
                id:  11 ,
                href: "\/bdi-viz-manual\/docs\/get-started\/",
                title: "Get Started",
                description: "Start your first BDIViz workflow",
                content: ""
            }
        );
    index.add(
            {
                id:  12 ,
                href: "\/bdi-viz-manual\/docs\/installation\/",
                title: "Installation",
                description: "Installation guide for setting up BDIViz on your system",
                content: ""
            }
        );
    index.add(
            {
                id:  13 ,
                href: "\/bdi-viz-manual\/docs\/manual\/",
                title: "User Manual",
                description: "Detailed user manual to BDIViz workflow",
                content: ""
            }
        );
    search.addEventListener('input', show_results, true);

    function show_results(){
        const maxResult =  5 ;
        const minlength =  0 ;
        var searchQuery = sanitizeHTML(this.value);
        var results = index.search(searchQuery, {limit: maxResult, enrich: true});

        
        const flatResults = new Map(); 
        for (const result of results.flatMap(r => r.result)) {
        if (flatResults.has(result.doc.href)) continue;
        flatResults.set(result.doc.href, result.doc);
        }

        suggestions.innerHTML = "";
        suggestions.classList.remove('d-none');

        
        if (searchQuery.length < minlength) {
            const minCharMessage = document.createElement('div')
            minCharMessage.innerHTML = `Please type at least <strong>${minlength}</strong> characters`
            minCharMessage.classList.add("suggestion__no-results");
            suggestions.appendChild(minCharMessage);
            return;
        } else {
            
            if (flatResults.size === 0 && searchQuery) {
                const noResultsMessage = document.createElement('div')
                noResultsMessage.innerHTML = "No results for" + ` "<strong>${searchQuery}</strong>"`
                noResultsMessage.classList.add("suggestion__no-results");
                suggestions.appendChild(noResultsMessage);
                return;
            }
        }

        
        for(const [href, doc] of flatResults) {
            const entry = document.createElement('div');
            suggestions.appendChild(entry);

            const a = document.createElement('a');
            a.href = href;
            entry.appendChild(a);

            const title = document.createElement('span');
            title.textContent = doc.title;
            title.classList.add("suggestion__title");
            a.appendChild(title);

            const description = document.createElement('span');
            description.textContent = doc.description;
            description.classList.add("suggestion__description");
            a.appendChild(description);

            suggestions.appendChild(entry);

            if(suggestions.childElementCount == maxResult) break;
        }
    }
    }());
</script>
        
    </body>
</html>
